Added WebSocket arg to allow manually setting port

ijjk opened the pull request:
Saw a reply on the original pull request that the WebSocket using a random port broke their set up so I added a `--websocket` or `-w` argument similar to the `-p` argument to allow manually setting this port also.

timneutkens commented:
Could someone that ran into this explain exactly why this wouldn't work? We're using Node.js defaults by listening on port 0. So it seems like the correct default / the user shouldn't need to worry about this 🤔

timneutkens commented:
Potentially because of a docker-compose type environment I guess?

ijjk commented:
Yeah, I would think using a random port would work for most setups using Next. Although being able to manually map it could be good in case anyone keeps their environment's firewall locked down on a port basis or they are using a docker-compose environment as you said.

@glenarama could you explain a little more your use case for manually mapping the WebSocket port?

glenarama commented:
I'm using a Chromebook for Dev, it has a Linux container which I use for Dev. 
In order to do local in browser testing/dev I need to open up specific ports. Currently the page reloads every few seconds because the port is blocked.

ijjk commented:
The `devWebSocketPort` name seems to make sense, I just pushed up that change.

timneutkens commented:
@ijjk could you add a section to the readme explaining that by default there is no need to configure this, but if you do you can do it by setting `devWebSocketPort`

timneutkens commented:
Could you also add a test for this if possible. There's a test suite for config options in `test/integration/config`

ijjk commented:
I added a note in the README under "Configuring the onDemandEntries" let me know if that's not the right place for it. I also added a test in the suite you mentioned. I had to dynamically write to the `next.config.js` for that test suite since there is the possibility of the port I define in the config being unavailable.

timneutkens commented on packages/next/server/hot-reloader.js:
```diff
@@ -173,8 +173,9 @@ export default class HotReloader {
+      const { devWebSocketPort } = this.config
       // create dynamic entries WebSocket
-      this.wss = new WebSocket.Server({ port: 0 }, function (err) {
+      this.wss = new WebSocket.Server({ port: devWebSocketPort || 0 }, function (err) {
```
Let's always read from `devWebsocketPort` and add `0` as a default in packages/next-server/server/config.js` Then we have a single source of truth 👍

ijjk replied:
Good idea, I updated it

timneutkens commented on packages/next/README.md:
```diff
@@ -1292,7 +1292,9 @@ module.exports = {
-  }
+  },
+  // optionally configure a port for the onDemandEntries WebSocket, not needed by default
+  devWebSocketPort: 3001,
```
I just realized we have a `onDemandEntries` key, totally forgot about it 🤦‍♂️ Can you move `devWebSocketPort` under `onDemandEntries` and call it `websocketPort` 🕵️

timneutkens replied:
Then I'll merge 💯

ijjk commented:
It's all good, that seems like a better place for it. I also moved the `onDemandEntries` config to the `defaultConfig` to keep that as the source of truth.

glenarama commented:
Thanks guys, really appreciate it!!

ijjk commented:
Happy New Year to you too!
